src/routines.cpp

```diff
@@ -74,6 +79,56 @@ SEXP xeusr_log(SEXP level_, SEXP msg_) {
     return R_NilValue;
 }
 
+SEXP xeusr_get_comm_manager__size() {
+    auto& manager = xeus_r::get_interpreter()->comm_manager();
+    return Rf_ScalarInteger(manager.comms().size());
+}
+
+SEXP comm_manager__register_target(SEXP name_) {
+    std::string name = CHAR(STRING_ELT(name_, 0));
+    
+    // TODO: for now the lambda is empty, because the callback is meant to be an 
+    //       R function stored on the R side, see routines.R/comm_target_env
+    xeus_r::get_interpreter()->comm_manager().register_comm_target(name, [name](xeus::xcomm&& comm, xeus::xmessage) {
+        Rprintf("open comm '%s' for target '%s'", comm.id().c_str(), name.c_str());
+    });
+    return R_NilValue;
+}
+
+SEXP comm_manager__unregister_target(SEXP name_) {
+    std::string name = CHAR(STRING_ELT(name_, 0));
+
+    xeus_r::get_interpreter()->comm_manager().unregister_comm_target(name);
+    return R_NilValue;
+}
+
+SEXP comm_manager__comm_open(SEXP s_comm_id, SEXP s_target_name, SEXP js_data) {
```

Comment:
Perhaps the comm_id should not come from the R side and be generated automatically with `new_xguid` ?